armory/logs.py

```diff
@@ -0,0 +1,228 @@
+        sink, format=format_log, level="TRACE", filter=filters, colorize=colorize
+    )
+    return new_logger
+
+
+def make_logfiles(output_dir: str) -> None:
+    """now that we have a output_dir start logging to the files there"""
+    global logfile_directory
+
+    logfile_directory = output_dir
+    log.trace(f"make_logfiles {output_dir}")
+    add_sink(f"{output_dir}/colored-log.txt", colorize=True),
+    add_sink(f"{output_dir}/armory-log.txt", colorize=False),
+
+
+# adapt the logging module to use the our loguru logger
+# this works by adding a new root handler which sends everything to
+# our InterceptHandler which extracts the extra information that loguru
+# needs and sends it to the loguru logger
+
+
+class InterceptHandler(logging.Handler):
+    def emit(self, record):
+        # Get corresponding Loguru level if it exists
+        try:
+            level = loguru.logger.level(record.levelname).name
+        except ValueError:
+            level = record.levelno
+
+        # Find caller from where originated the logged message
+        frame, depth = logging.currentframe(), 2
+        while frame.f_code.co_filename == logging.__file__:
+            frame = frame.f_back
+            depth += 1
+
+        loguru.logger.opt(depth=depth, exception=record.exc_info).log(
+            level, record.getMessage()
+        )
+
+
+def log_method(entry=True, exit=True, level="DEBUG"):
+    """function decorator that logs entry and exit"""
+
+    def wrapper(func):
+        name = func.__name__
+
+        @functools.wraps(func)
+        def wrapped(*args, **kwargs):
+            logger_ = log.opt(depth=1)
+            if entry:
+                logger_.log(
+                    level, "Entering '{}' (args={}, kwargs={})", name, args, kwargs
+                )
+            result = func(*args, **kwargs)
+            if exit:
+                logger_.log(level, "Exiting '{}' (result={})", name, result)
+            return result
+
+        return wrapped
+
+    return wrapper
+
+
+logging.basicConfig(handlers=[InterceptHandler()], level=0)
+
+if __name__ == "__main__":
+    update_filters(["armory:INFO", "art:INFO"])
+    for level in "trace debug info success warning error critical".split():
+        log.log(level.upper(), f"message at {level}")
+
+    def library_messages():
+        # this should induce a flurry of log messages through the InterceptHandler
+        import boto3
+
+        s3 = boto3.client("s3")
+        s3.get_bucket_location(Bucket="armory-submission-data")
+
+        import tensorflow as tf
+
+        (x_train, y_train), (x_test, y_test) = tf.keras.datasets.mnist.load_data()
+
+        log.success("dev test complete")
+
+    log.info("logging library messages at INFO")
+    library_messages()
+    update_filters(["s3transfer:DEBUG"])
+    log.info("logging s3:DEBUG")
+    library_messages()
+
+# need to instantiate the filters early, replacing the default
+log.remove()
+add_sink(sys.stdout)
```

Comment:
Is there an easy way for interactive / jupyter users to modify logging levels after importing armory? E.g., something like:
```
import armory.logs
armory.logs.set_level("warning")
```
Maybe the following:
```
armory.logs.update_filters(["armory:warning"])
```
?